### PR TITLE
Use Cmake to find Python and Sphinx-doc

### PR DESCRIPTION
**Pull request by xvitaly**
Use Cmake to find Python and Sphinx-doc.

I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: doc/CMakeLists.txt [hunk 1]
```diff
@@ -4,10 +4,17 @@ if (NOT DOXYGEN)
   return ()
 endif ()
 
+find_package(PythonInterp
+  QUIET REQUIRED)
+find_program(SPHINX_EXECUTABLE
+  NAMES sphinx-build sphinx-build-3)
```
Comment (vitaut) [inline doc/CMakeLists.txt:10]: Let's put arguments on the same line since the command is short.

File: doc/CMakeLists.txt [hunk 2]
```diff
@@ -4,10 +4,17 @@ if (NOT DOXYGEN)
...
+
 add_custom_target(doc
-  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/build.py ${FMT_VERSION}
+  COMMAND ${CMAKE_COMMAND} -E env SPHINX_EXECUTABLE=${SPHINX_EXECUTABLE}
```
Comment (vitaut) [inline doc/CMakeLists.txt:13]: Let's invoke `PYTHON_EXECUTABLE` directly, not through cmake and pass `SPHINX_EXECUTABLE` as an optional command-line argument.

**Comment by xvitaly** [comment]
Done.

**Comment by vitaut** [comment]
Merged in https://github.com/fmtlib/fmt/commit/638db5ca5e47066b0b9ac297bdcfcae69ccd0276, thanks!

**Pull request closed without merging**